File: solutions/uri/1329/1329.cpp
#include <iostream>

int
    main() {
    int n, c, m, j;

    while (std::cin >> n) {
        m = 0;
        j = 0;

        if (n == 0)
            break;

        while (n--) {
            std::cin >> c;

            if (c == 0) {
                m++;
            } else {
                j++;
            }
        }

        std::cout << "Mary won " << m << " times and John won " << j;
        std::cout << " times" << std::endl;
    }

    return 0;
}

File: solutions/beecrowd/1329/1329.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

while line := input().strip():
    n = int(line)
    if n == 0:
        break

    results = list(map(int, input().split()))

    m = j = 0
    for result in results:
        if result == 0:
            m += 1
        else:
            j += 1

    output(f'Mary won {m} times and John won {j} times\n')
